### PR TITLE
Add unit tests for bat-native-confirmations token redemption

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/9288

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

- Confirm "Estimated rewards pending" shows the correct value for ads rewards and balances after tokens have been redeemed

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: vendor/bat-native-confirmations/src/bat/confirmations/internal/ad_grants.cc [hunk 1]
```diff
@@ -45,12 +45,13 @@ bool AdGrants::SetFromJson(const std::string& json) {
 bool AdGrants::SetFromDictionary(base::DictionaryValue* dictionary) {
   DCHECK(dictionary);
 
-  auto* value = dictionary->FindKey("grants_balance");
-  if (!value || !value->is_double()) {
+  const base::Optional<double> balance =
```
Comment (tmancey) [inline vendor/bat-native-confirmations/src/bat/confirmations/internal/ad_grants.cc:48]: Unit tests highlighted that `0` in the `confirmations.json` state file for the `balance` would return `false` due to `is_double()` failing

File: vendor/bat-native-confirmations/src/bat/confirmations/internal/payments.cc [hunk 2]
```diff
@@ -65,12 +65,13 @@ bool Payments::SetFromDictionary(base::DictionaryValue* dictionary) {
     PaymentInfo payment;
 
     // Balance
-    auto* balance_value = dictionary->FindKey("balance");
-    if (!balance_value || !balance_value->is_double()) {
+    const base::Optional<double> balance_value =
```
Comment (tmancey) [inline vendor/bat-native-confirmations/src/bat/confirmations/internal/payments.cc:68]: Unit tests highlighted that `0` in the `confirmations.json` state file for the `balance` would return `false` due to `is_double()` failing

File: vendor/bat-native-confirmations/src/bat/confirmations/internal/confirmations_unblinded_tokens_unittest.cc [hunk 3]
```diff
@@ -3,41 +3,40 @@
...
 #include <string>
 #include <vector>
-#include <memory>
-#include <fstream>
-#include <sstream>
+
+    MockLoadState(confirmations_client_mock_.get());
+    MockSaveState(confirmations_client_mock_.get());
+
+    Initialize(confirmations_.get());
```
Comment (emerick) [inline vendor/bat-native-confirmations/src/bat/confirmations/internal/confirmations_unblinded_tokens_unittest.cc:13]: Is this meant to be here?

Comment (tmancey) [reply vendor/bat-native-confirmations/src/bat/confirmations/internal/confirmations_unblinded_tokens_unittest.cc]: nope! ❤️

Comment (tmancey) [reply vendor/bat-native-confirmations/src/bat/confirmations/internal/confirmations_unblinded_tokens_unittest.cc]: Fixed and re-run tests

**Comment by tmancey** [comment]
Windows x64 failed CI, restarting

**Comment by tmancey** [comment]
macOS and Windows x64 failed CI, restarting

**Comment by tmancey** [comment]
Windows x64 failed CI, restarting

**Comment by tmancey** [comment]
CI passed at https://ci.brave.com/job/brave-browser-build-pr/job/issues%252F9288/

**Pull request merged**